src/main/java/net/neoforged/neoforge/transfer/ResourceHandlerUtil.java

```diff
@@ -548,44 +524,76 @@ public static <T extends IResource> int indexOf(IResourceHandler<T> handler, T r
         return -1;
     }
 
-    // TODO: unsafe access to tx state
-//    public static <T extends IResource> boolean hasExtractableResource(IResourceHandler<T> handler, Predicate<T> filter) {
-//        try (Transaction temp = UnsafeTransactionManager.openUnsafe()) {
-//            //Simulated: we don't commit
-//            int size = handler.size();
-//            for (int index = 0; index < size; index++) {
-//                T resource = handler.getResource(index);
-//                if (!doesNotMatch(filter, resource) && handler.extract(resource, 1, temp) > 0)
-//                    return true;
-//            }
-//            return false;
-//        }
-//    }
-
-    // TODO: unsafe access to tx state
-//    public static <T extends IResource> boolean hasExtractableResourceAtIndex(IResourceHandler<T> handler, Predicate<T> filter, int index) {
-//        try (Transaction temp = UnsafeTransactionManager.openUnsafe()) {
-//            //Simulated: we don't commit
-//            T resource = handler.getResource(index);
-//            return !doesNotMatch(filter, resource) && handler.extract(resource, 1, temp) > 0;
-//        }
-//    }
-//
-//    public static <T extends IResource> boolean hasExtractableResource(IResourceHandler<T> handler, T resource) {
-//        try (Transaction temp = UnsafeTransactionManager.openUnsafe()) {
-//            //Simulated: we don't commit
-//            return handler.extract(resource, 1, temp) > 0;
-//        }
-//    }
-
-    // TODO: do we want it?
-//    public static <T extends IResource> T getFirstResourceOrDefault(IResourceHandler<T> handler, T defaultResource) {
-//        int size = handler.size();
-//        for (int index = 0; index < size; index++) {
-//            T resource = handler.getResource(index);
-//            if (!resource.isEmpty())
-//                return resource;
-//        }
-//        return defaultResource;
-//    }
+    /**
+     * Checks if the given resource handler has at least one resource that matches the given filter and can be extracted.
```

Comment:
Let's instead call it `findExtractableResource`, and make it return the resource it found? I have a use case for this in MI's pipes. (i.e. finding an extractable resource to assign the fluid for the pipe network, if no fluid was assigned yet).